### ISSUE TITLE
Unexpected behavior when scheduling different learning rates

### ISSUE THREAD

**Issue by nateagr** [tensorflow/tensorflow#28989]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Centos7
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): Binary
- TensorFlow version (use command below): v1.12.0-0-ga6d8ffae09 1.12.0
- Python version: 3.6
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

I've implemented a custom logistic regression using Estimator API which gives good results when learning with a fixed set of hyperparameters. However, I'm getting unexpected results when learning the model with a learning rate which varies over time. As you can see in the screenshot below, variables of the model (weights and bias), losses and metric prediction_mean abruptly changed twice during the learning, coinciding with learning rate modification.

Here is the piece of code describing instanciation of the estimator:
```
tf.estimator.Estimator(
    model_fn=my_model_fn,
    params={
        "optimizer": lambda: tf.train.FtrlOptimizer(
            learning_rate=tf.where(
                tf.less(tf.train.get_global_step(), 100000),
                3e-3,
                tf.where(
                    tf.less(tf.train.get_global_step(), 200000),
                    9e-4,
                    tf.where(
                        tf.less(tf.train.get_global_step(), 300000),
                        6e-4,
                        3e-4
                    )
                )
            ),
            l2_regularization_strength=1e-2
        )
    },
    model_dir=model_dir
)
```

![image](https://user-images.githubusercontent.com/41445208/58311641-b8b22980-7e09-11e9-9460-a33cd94f713e.png)

I was wondering if the modification of the learning rate could trigger the modification of the state of FTRL which could explain this behavior. Any clue ? Thanks a lot in advance.

**Comment by achandraa** [comment]
Can you please provide full minimal code snippet that can reproduce the issue at hand. This would help us to dig in easily. Thanks!

**Comment by achandraa** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
